hibernate-core/src/main/java/org/hibernate/internal/SessionImpl.java

```diff
@@ -2969,6 +2969,11 @@ public boolean isSessionCheckingEnabled() {
 			return sessionCheckingEnabled;
 		}
 
+		@Override
+		public boolean isSessionFactoryCheckingEnabled() {
```

Comment:
Just like the 1st-level cache is linked to the Session, the 2nd-level cache is linked to the SessionFactory. The name of the method was chosen to resemble the `isSessionCheckingEnabled` method we already have.

However, since we already have methods like `loadFromSessionCache` and `loadFromSecondLevelCache`, it's probably better to use `SecondLevelCache` in this context.